fe/fe-core/src/main/java/com/starrocks/sql/optimizer/SubqueryUtils.java

```diff
@@ -19,12 +24,69 @@
 import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.operator.scalar.SubqueryOperator;
+import com.starrocks.sql.optimizer.rewrite.ScalarOperatorRewriter;
+import com.starrocks.sql.optimizer.rewrite.scalar.ReplaceSubqueryRewriteRule;
+import com.starrocks.sql.optimizer.rewrite.scalar.ScalarOperatorRewriteRule;
+import com.starrocks.sql.optimizer.transformer.CTETransformerContext;
+import com.starrocks.sql.optimizer.transformer.ExpressionMapping;
+import com.starrocks.sql.optimizer.transformer.LogicalPlan;
+import com.starrocks.sql.optimizer.transformer.OptExprBuilder;
+import com.starrocks.sql.optimizer.transformer.RelationTransformer;
 
 import java.util.List;
 import java.util.Map;
 
 public class SubqueryUtils {
 
+    public static Pair<ScalarOperator, OptExprBuilder> rewriteScalarOperator(
+            ScalarOperator scalarOperator,
+            OptExprBuilder builder,
+            Map<ScalarOperator, SubqueryOperator> subqueryPivots) {
```

Comment:
In all places, seems no Pivot meaning, only is a placeholder.

Reply:
Ok, seems a better naming